src/tests.ts

```diff
@@ -4,6 +4,20 @@ import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
 import "regenerator-runtime/runtime"
 import { format } from "util"
 import "@testing-library/jest-dom"
+import { configure } from "@testing-library/react"
+
+configure({
+  // Fixes issue where testing-library react await would time out on CI
+  asyncUtilTimeout: 10000,
+  // Don't spit out html output on failed tests by default. If needing to see
+  // html, use screen.debug(undefined, Infinity)
```

Comment:
How about a compromise: [a local ENV var](https://github.com/artsy/force/blob/main/.env.example)? HTML output is _always_ truncated, except for in the smallest of components. We can expand the output via an ENV var, but even then, the general gist of the test sequence / file is lost in all of the noise.

Reply:
Yep, a local env var works for me, something like:

```ts
if (
  process.env.TERSE_RTL_OUTPUT === "true" ||
  process.env.CIRCLECI === "true"
) {
  configure({
    // same as above
  })
 }
```

BUT I think it should definitely be an **opt-in** rather than opt-out since it is such an aggressive departure from expected RTL behavior.

See below for example of how much useful detail is lost when iterating locally.

<details>
 <summary>Example of terse output</summary>

<pre>
❯ TERSE_RTL_OUTPUT=true yarn jest src/Components/SavedSearchAlert/__tests__/ConfirmationArtworksGrid.jest.tsx

FAIL  src/Components/SavedSearchAlert/__tests__/ConfirmationArtworksGrid.jest.tsx
  ConfirmationArtworksGrid
    ✕ renders artworks count and artworks grid (53 ms)
    ✓ displays the correct message when there are many matches (10 ms)
    ✓ displays the correct message when there are few matches (8 ms)
    ✓ displays the correct message when there are is one match (7 ms)
    ✓ displays the correct message when there are no matches (3 ms)

  ● ConfirmationArtworksGrid › renders artworks count and artworks grid

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        2.657 s
</pre>
</details>

<details>
 <summary>Example of regular output</summary>

<pre>
❯ yarn jest src/Components/SavedSearchAlert/__tests__/ConfirmationArtworksGrid.jest.tsx

FAIL  src/Components/SavedSearchAlert/__tests__/ConfirmationArtworksGrid.jest.tsx
  ConfirmationArtworksGrid
    ✕ renders artworks count and artworks grid (57 ms)
    ✓ displays the correct message when there are many matches (9 ms)
    ✓ displays the correct message when there are few matches (7 ms)
    ✓ displays the correct message when there are is one match (7 ms)
    ✓ displays the correct message when there are no matches (3 ms)

  ● ConfirmationArtworksGrid › renders artworks count and artworks grid

    ⚠️👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽  Key information being lost!   👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽⚠️
    TestingLibraryElementError: Unable to find an element with the text: See our top picks for you!. This could be because the text is broken up by multiple elements. In this case, you can provide a function for your text matcher to make your matcher more flexible.

    &lt:body&gt;
      &lt:div&gt;
        &lt:div
          class="Box-sc-15se88d-0 Flex-cw39ct-0 jkcvUH"
          display="flex"
        &gt;
          &lt:div
            class="Box-sc-15se88d-0 Text-sc-18gcpao-0 ibSBu eRhYMG"
            color="black60"
            font-family="sans"
          &gt;
            300 works currently on Artsy match your criteria.
          &lt:/div&gt;
          &lt:div
            class="Box-sc-15se88d-0 Text-sc-18gcpao-0 ibSBu eRhYMG"
            color="black60"
            font-family="sans"
          &gt;
            ⚠️👇🏽  Relevant markup, of course  👇🏽⚠️
            See our top picks for you:
          &lt:/div&gt;
          &lt:div
            class="Box-sc-15se88d-0 fMtsRi"
          /&gt;
          &lt:div
            class="Box-sc-15se88d-0 CSSGrid-sc-1q8w5xn-0 GridColumns-sc-1g9p6xx-0  hiilRS fmlhQs"
          &gt;
            &lt:div
              class="Box-sc-15se88d-0 GridColumns__Cell-sc-1g9p6xx-1  eoBGTk"
            &gt;
              &lt:div
                class="ArtworkGrid-uextjn-0 fSsgxk"
                data-test="artworkGrid"
              &gt;
                &lt:div
                  style="flex: 1; min-width: 0; margin-right: 20px;"
                &gt;
                  &lt:div
                    data-id="&lt:Artwork-mock-id-3&gt;"
                    data-test="artworkGridItem"
                    relay="[object Object]"
                  &gt;
                    &lt:div
                      class="Box-sc-15se88d-0 gKyINF"
                      width="100%"
                    &gt;
                      &lt:a
                        aria-label="Happy Choppers by &lt:mock-value-for-field-\"artistNames\"&gt;"
                        class="RouterLink__RouterUnawareLink-sc-1nwbtp5-1 fhtVkp GridItem__Link-d2n1vy-0 jnqBCS"
                        data-testid="artwork-link"
                        href="&lt:mock-value-for-field-\"href\"&gt;"
                      &gt;
                        &lt:div
                          class="Box-sc-15se88d-0 fIeAnl"
                          height="100%"
                          overflow="hidden"
                          width="100%"
                        &gt;
                          &lt:div
                            class="Box-sc-15se88d-0 Skeleton__SkeletonBox-sc-1vwqe5c-0 hlBmjG MagnifyImage__Image-sc-1vv0ilh-0 gEgKhU"
                          &gt;
                            &lt:img
                              alt="&lt:mock-value-for-field-\"image_title\"&gt;"
                              class="LazyImage__InnerLazyImage-sc-1fxlbs3-0 ldtabI"
                              src="undefined?height=105&quality=80&resize_to=fit&src=%3Cmock-value-for-field-%22url%22%3E&width=445"
                              srcset="undefined?height=105&quality=80&resize_to=fit&src=%3Cmock-value-for-field-%22url%22%3E&width=445 1x, undefined?height=210&quality=80&resize_to=fit&src=%3Cmock-value-for-field-%22url%22%3E&width=890 2x"
                              style="transform-origin: center center; transition: transform 0.15s ease, transform-origin 100ms ease, opacity 0.25s; transform: scale(1.0); opacity: 0;"
                            /&gt;
                            &lt:noscript /&gt;
                          &lt:/div&gt;
                        &lt:/div&gt;
                      &lt:/a&gt;
                    &lt:/div&gt;
                    &lt:a
                      class="RouterLink__RouterUnawareLink-sc-1nwbtp5-1 kwtXtv"
                      data-testid="metadata-artwork-link"
                      display="block"
                      href="&lt:mock-value-for-field-\"href\"&gt;"
                      text-decoration="none"
                    &gt;
                      &lt:div
                        class="Box-sc-15se88d-0"
                      &gt;
                        &lt:div
                          class="Box-sc-15se88d-0 Flex-cw39ct-0 gaIPKD"
                          display="flex"
                        &gt;
                          &lt:div
                            class="Box-sc-15se88d-0 Text-sc-18gcpao-0  cVDWrC"
                            font-family="sans"
                          &gt;
                            &lt;mock-value-for-field-"cultural_maker"&gt;
                          &lt:/div&gt;
                          &lt:button
                            aria-label="Save"
                            class="Clickable-sc-10cr82y-0 iigefC"
                            cursor="pointer"
                            data-test="saveButton"
                            height="20"
                            type="button"
                            width="20"
                          &gt;
                            &lt:div
                              class="sc-gtsrHT gZriJc"
                              fill="black100"
                              height="20"
                              title="Save"
                              width="20"
                            &gt;
                              &lt:svg
                                fill="currentColor"
                                style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px; width: 100%; height: 100%;"
                                viewBox="0 0 18 18"
                              &gt;
                                &lt:path
                                  d="M11.9998 3.00002C11.4743 2.9996 10.954 3.10272 10.4684 3.30347C9.9828 3.50422 9.54153 3.79868 9.16978 4.17002L8.99978 4.34002L8.82978 4.17002C8.07922 3.41945 7.06124 2.99779 5.99978 2.99779C4.93833 2.99779 3.92035 3.41945 3.16978 4.17002C2.41922 4.92058 1.99756 5.93856 1.99756 7.00002C1.99756 8.06147 2.41922 9.07945 3.16978 9.83002L8.64978 15.3C8.69467 15.3478 8.74889 15.386 8.80909 15.412C8.86929 15.4381 8.93419 15.4515 8.99978 15.4515C9.06538 15.4515 9.13028 15.4381 9.19048 15.412C9.25068 15.386 9.30489 15.3478 9.34978 15.3L14.8298 9.83002C15.3898 9.27059 15.7713 8.55758 15.9259 7.78124C16.0805 7.0049 16.0013 6.20015 15.6983 5.46886C15.3953 4.73756 14.8821 4.11262 14.2237 3.67313C13.5653 3.23365 12.7914 2.99939 11.9998 3.00002ZM14.1198 9.12002L8.99978 14.24L3.87978 9.12002C3.58504 8.84537 3.34863 8.51417 3.18466 8.14617C3.02069 7.77817 2.93252 7.38092 2.92542 6.97811C2.91831 6.57529 2.99241 6.17518 3.14329 5.80163C3.29418 5.42807 3.51875 5.08874 3.80363 4.80386C4.0885 4.51899 4.42784 4.29441 4.80139 4.14353C5.17495 3.99264 5.57506 3.91854 5.97787 3.92565C6.38068 3.93276 6.77794 4.02092 7.14594 4.18489C7.51393 4.34886 7.84513 4.58527 8.11978 4.88002L8.64978 5.40002L8.99978 5.76002L9.34978 5.40002L9.87978 4.88002C10.4485 4.3501 11.2007 4.0616 11.9779 4.07532C12.7551 4.08903 13.4966 4.40388 14.0463 4.95353C14.5959 5.50318 14.9108 6.24472 14.9245 7.02193C14.9382 7.79913 14.6497 8.55132 14.1198 9.12002Z"
                                /&gt;
                              &lt:/svg&gt;
                            &lt:/div&gt;
                          &lt:/button&gt;
                        &lt:/div&gt;
                        &lt:div
                          class="Box-sc-15se88d-0 jGXstL"
                        &gt;
                          &lt:span
                            to="&lt:mock-value-for-field-\"href\"&gt;"
                          &gt;
                            &lt:div
                              class="Box-sc-15se88d-0 Text-sc-18gcpao-0 ibSBu RgmAL"
                              color="black60"
                              font-family="sans"
                            &gt;
                              &lt:i&gt;
                                Happy Choppers
                              &lt:/i&gt;
                              , &lt;mock-value-for-field-"date"&gt;
                            &lt:/div&gt;
                          &lt:/span&gt;
                          &lt:div
                            class="Box-sc-15se88d-0 Text-sc-18gcpao-0 ibSBu heSeEW"
                            color="black60"
                            font-family="sans"
                          &gt;
                            &lt;mock-value-for-field-"collecting_institution"&gt;
                          &lt:/div&gt;
                        &lt:/div&gt;
                        &lt:div
                          class="Box-sc-15se88d-0 Text-sc-18gcpao-0 fJIzlB kduJmf"
                          color="black100"
                          font-family="sans"
                          font-weight="bold"
                        &gt;
                          &lt;mock-value-for-field-"sale_message"&gt;
                           
                        &lt:/div&gt;
                      &lt:/div&gt;
                    &lt:/a&gt;
                  &lt:/div&gt;
                &lt:/div&gt;
                &lt:div
                  style="flex: 1; min-width: 0; margin-right: 0px;"
                /&gt;
              &lt:/div&gt;
            &lt:/div&gt;
          &lt:/div&gt;
          &lt:div
            class="Box-sc-15se88d-0 fMtsRi"
          /&gt;
        &lt:/div&gt;
      &lt:/div&gt;
    &lt:/body&gt;

      41 |       screen.getByText("300 works currently on Artsy match your criteria.")
      42 |     ).toBeInTheDocument()
    ⚠️👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽  The failed assertion itself 👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽👇🏽⚠️
    &gt; 43 |     expect(screen.getByText("See our top picks for you!")).toBeInTheDocument()
         |                   ^
      44 |   })
      45 |
      46 |   it("displays the correct message when there are many matches", async () =&gt; {

      at Object.getElementError (node_modules/@testing-library/dom/dist/config.js:37:19)
      at node_modules/@testing-library/dom/dist/query-helpers.js:90:38
      at node_modules/@testing-library/dom/dist/query-helpers.js:62:17
      at node_modules/@testing-library/dom/dist/query-helpers.js:111:19
      at Object.getByText (src/Components/SavedSearchAlert/__tests__/ConfirmationArtworksGrid.jest.tsx:43:19)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        2.605 s, estimated 3 s
</pre>
</details>

Reply:
The vast majority of the time this error message does not give me helpful information in the truncated HTML. A large test suite with multiple failures it might not even be possible to scroll up and see which tests actually failed. So I generally support this change, but if there are things we can do to be more clever or strike and agreeable balance, That’s OK too. Would it make sense to try and catch the failure  part of the error message and drop the markup?

Reply:
My 2¢ is that the very quiet output configured here is totally fine for many situations, for all the reasons you've cited — and my suggestion would absolutely allow for that  👍🏽

It's totally possible that the components in the commerce flows have the kind of complexity that warrant this, no argument there!

But for me (and perhaps others?) this is a net loss and would be unexpected / hard to track down, which is why I think it should be opt-in.

Reply:
I.e. as simple as `TERSE_RTL_OUTPUT=true yarn test` 

If I'm in the minority, happy to flip that env var to an opt-out `VERBOSE_RTL_OUTPUT=true yarn test` but I don't think we actually know where folks stand yet?

Reply:
In general, opt-out's are more ideal than opt in, especially for departures from expected behavior. So yah 👍

Reply:
Just curious if you folks considered the solution to reducing RTL output instead([reference](https://testing-library.com/docs/dom-testing-library/api-debugging/)), https://testing-library.com/docs/dom-testing-library/api-debugging/ 

If the problem is the big output setting a super low value for this var would work in the CI.

Reply:
Its more about making this feature useful. In many cases its not useful at all. In other small / isolated cases it is. Mostly, things seem to truncate, something which can be fixed with `debug(undefined, Infinity)`. To make it by default really long will just make things even more lost in larger test sets where > 1 test is being run